tutorials/blink_led.rst

```diff
@@ -0,0 +1,378 @@
+
+    @external(erlang, "gpio", "set_pin_mode")
+    pub fn gpio_set_pin_mode(pin: Int, direction: Direction) -> Nil
+
+    @external(erlang, "gpio", "digital_write")
+    pub fn gpio_digital_write(pin: Int, level: Int) -> Nil
+
+    pub fn start() {
+        gpio_init(pin)
+        gpio_set_pin_mode(pin, Output)
+        blink_loop(1)
+    }
+
+    fn blink_loop(level) {
+        gpio_digital_write(pin, level)
+        process.sleep(1000)
+        blink_loop(1 - level)
+    }
+
+Building, flashing and running on Raspberry Pi Pico
+---------------------------------------------------
+
+To program the Raspberry Pi Pico, you need to press the button and connect it,
+and then write an .uf2 file on the mounted drive.
+
+.. tab:: Erlang
+
+    .. code-block:: bash
+
+        cd blink_led
+        rebar3 atomvm pico_flash
+
+    rebar3, with AtomVM plugin, automatically builds, packs (to .avm), creates the uf2, puts the Pico in the proper mode and copies the firmware at the proper location. It doesn't hurt to do the steps one after the other if you prefer.
+
+    .. code-block:: bash
+
+        rm -rf _build
+        rebar3 get-deps
+        rebar3 compile
+        rebar3 atomvm packbeam
+        rebar3 atomvm uf2create
+        rebar3 atomvm pico_flash
+
+    If there are some errors such as `"===> Pico not mounted at /Volumes/RPI-RP2."` on macOS, unplug the Pico, press the button, plug it (maybe on another USB port) and check that indeed `/Volumes/RPI-RP2` exists. The equivalent path on Linux is `/run/media/${USER}/RPI-RP2`.
+
+    Also, macOS will display an alert `"Disk Not Ejected Properly"`. This is a known issue between Raspberry Pi and Apple. You can just ignore it.
+
+.. tab:: Elixir
+
+    With mix, you need first to install dependencies:
+
+    .. code-block:: bash
+
+        cd blink_led
+        mix deps.get
+
+    Then you can build and flash with:
+
+    .. code-block:: bash
+
+        mix atomvm.pico.flash
+
+    You can safely ignore the warnings about GPIO module being not available for this tutorial, as well as the warning about `avm_deps` directory being not found, as the tutorial code doesn't use any Erlang or Elixir module. But if you were to use Elixir modules, you would need to create an avm_deps directory for your .avm file to embed Elixir modules not present in the lib partition.
+
+    mix, with ExAtomVM plugin, automatically builds, packs (to .avm), creates the uf2, puts the Pico in the proper mode and copies the firmware at the proper location. It doesn't hurt to do the steps one after the other if you prefer.
+
+    .. code-block:: bash
+
+        rm -rf _build
+        mix compile
+        mix atomvm.packbeam
+        mix atomvm.uf2create
+        mix atomvm.pico.flash
+
+.. tab:: Gleam
+
+    The first step is to compile everything as an erlang-shipment so we know all .beam files are generated.
+
+    .. code-block:: bash
+
+        cd blink_led
+        gleam export erlang-shipment
```

Comment:
This is a good start, and based on the information you have included above gleam in this tutorial I have some ideas about improving the gleam experience with some enhancement to the atomvm_rebar3 plugin.

Do you happen to know if there is a way to accomplish this natively from the BEAM with a functional call to a gleam module? That seems preferable to using os cmd from within the rebar plugin.